### ISSUE TITLE
NoSuchMethodError: The getter 'attributes' was called on null.

### ISSUE THREAD

**Issue by SwissCheese5** [Hexer10/youtube_explode_dart#29]
```
NoSuchMethodError: The getter 'attributes' was called on null.
Receiver: null
Tried calling: attributes
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      YoutubeExplode.getVideo (package:youtube_explode_dart/src/youtube_explode_base.dart:405:10)
<asynchronous suspension>
#2      <my processor class>
#3      main.<anonymous closure>.<anonymous closure> (<my class>)
#4      Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:172:27)
<asynchronous suspension>
#5      Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#6      Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:246:15)
#7      _rootRun (dart:async/zone.dart:1184:13)
#8      _CustomZone.run (dart:async/zone.dart:1077:19)
#9      _runZoned (dart:async/zone.dart:1619:10)
#10     runZoned (dart:async/zone.dart:1539:10)
#11     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:243:5)
#12     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:170:33)
#13     _rootRun (dart:async/zone.dart:1184:13)
#14     _CustomZone.run (dart:async/zone.dart:1077:19)
#15     _runZoned (dart:async/zone.dart:1619:10)
#16     runZoned (dart:async/zone.dart:1539:10)
#17     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:13)
#18     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:400:30)
#19     new Future.<anonymous closure> (dart:async/future.dart:176:37)
#20     StackZoneSpecification._run (package:stack_trace/src/stack_zone_specification.dart:209:15)
#21     StackZoneSpecification._registerCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:119:48)
#22     _rootRun (dart:async/zone.dart:1180:38)
#23     _CustomZone.run (dart:async/zone.dart:1077:19)
#24     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
#25     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
#26     StackZoneSpecification._run (package:stack_trace/src/stack_zone_specification.dart:209:15)
#27     StackZoneSpecification._registerCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:119:48)
#28     _rootRun (dart:async/zone.dart:1184:13)
#29     _CustomZone.run (dart:async/zone.dart:1077:19)
#30     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1003:23)
#31     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:23:15)
#32     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19)
#33     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
#34     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
Expected: <true>
  Actual: <false>
#0      fail (package:test_api/src/frontend/expect.dart:154:30)
#1      _expect (package:test_api/src/frontend/expect.dart:148:3)
#2      expect (package:test_api/src/frontend/expect.dart:60:3)
#3      main.<anonymous closure>.<anonymous closure> (<my class>)
<asynchronous suspension>
#4      main.<anonymous closure>.<anonymous closure> (<my class>)
#5      Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:172:27)
<asynchronous suspension>
#6      Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#7      Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:246:15)
#8      _rootRun (dart:async/zone.dart:1184:13)
#9      _CustomZone.run (dart:async/zone.dart:1077:19)
#10     _runZoned (dart:async/zone.dart:1619:10)
#11     runZoned (dart:async/zone.dart:1539:10)
#12     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:243:5)
#13     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:170:33)
#14     _rootRun (dart:async/zone.dart:1184:13)
#15     _CustomZone.run (dart:async/zone.dart:1077:19)
#16     _runZoned (dart:async/zone.dart:1619:10)
#17     runZoned (dart:async/zone.dart:1539:10)
#18     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:13)
#19     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:400:30)
#20     new Future.<anonymous closure> (dart:async/future.dart:176:37)
#21     StackZoneSpecification._run (package:stack_trace/src/stack_zone_specification.dart:209:15)
#22     StackZoneSpecification._registerCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:119:48)
#23     _rootRun (dart:async/zone.dart:1180:38)
#24     _CustomZone.run (dart:async/zone.dart:1077:19)
#25     _CustomZone.runGuarded (dart:async/zone.dart:979:7)
#26     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1019:23)
#27     StackZoneSpecification._run (package:stack_trace/src/stack_zone_specification.dart:209:15)
#28     StackZoneSpecification._registerCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:119:48)
#29     _rootRun (dart:async/zone.dart:1184:13)
#30     _CustomZone.run (dart:async/zone.dart:1077:19)
#31     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1003:23)
#32     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:23:15)
#33     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19)
#34     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
#35     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

```dart
      var id = YoutubeExplode.parseVideoId(url);

      var video = await YoutubeExplode().getVideo(id);
```
This is in my test file. This was working with the exact same url for over 50 times before. It suddenly stopped working today without changing any code.

**Comment by SwissCheese5** [comment]
This worked after re-running the test again, no code changes.
This seems to be a rare edge-case.

**Comment by phibr0** [comment]
I have this issues sometimes too..

**Comment by Hexer10** [comment]
This is be fixed in `1.0.0-beta`, you can already try it, but is still missing some features see( https://github.com/Hexer10/youtube_explode_dart/milestone/1 ), and has also some breaking changes, you can see how to library syntax change in the v5 branch which includes the updated README and examples, also check the pub.dev documentation: https://pub.dev/documentation/youtube_explode_dart/1.0.0-beta/
